daemon/src/taker_cfd.rs

```diff
@@ -167,6 +185,65 @@ impl<O, M, W> Actor<O, M, W> {
 
         Ok(())
     }
+
+    async fn handle_auto_rollover(&mut self) -> Result<()> {
+        tracing::debug!("Auto rollover");
+
+        let mut conn = self.db.acquire().await?;
+        let cfds = load_all_cfds(&mut conn).await?;
+
+        // cleanup all pending proposals because auto-rollover will create a new one
+        self.current_pending_proposals.clear();
+
+        // Only cfds in state `Open` are allowed for auto-rollover proposal
+        for cfd in cfds
+            .iter()
+            .filter(|cfd| matches!(cfd.state, CfdState::Open { .. }))
+        {
+            let now = OffsetDateTime::now_utc();
+
+            let elapsed_since_creation =
+                now - (cfd.expiry_timestamp() - self.roll_over_params.settlement_interval);
+
+            // Two conditions need to be bet for a roll over to be triggered.
+            // The cfd should not of run longer that the settlement interval.
+            // The cfd should of lived longer than the rollover interval.
+            if self.roll_over_params.settlement_interval > elapsed_since_creation
+                && elapsed_since_creation < self.roll_over_params.rollover_interval
+            {
+                try_continue!(self.propose_roll_over(cfd.order.id).await)
+            }
+        }
+
+        Ok(())
+    }
+
+    async fn propose_roll_over(&mut self, order_id: OrderId) -> Result<()> {
+        let proposal = RollOverProposal {
+            order_id,
+            timestamp: Timestamp::now()?,
+        };
+
+        self.current_pending_proposals.insert(
+            proposal.order_id,
+            UpdateCfdProposal::RollOverProposal {
+                proposal: proposal.clone(),
+                direction: SettlementKind::Outgoing,
+            },
+        );
+        self.send_pending_update_proposals()?;
+
+        self.send_to_maker
+            .send(wire::TakerToMaker::ProposeRollOver {
+                order_id: proposal.order_id,
+                timestamp: proposal.timestamp,
+            })
+            .await?;
+
+        tracing::debug!("proposed roll over");
```

Comment:
I think this is important enough to warrant an `info` log, plus it would be good to include some context.

```suggestion
        tracing::info!(%order_id, "Proposed roll over");
```